### PR TITLE
silo: Fix HDF5 1.13 API breakage

### PR DESCRIPTION
**Pull request by chuckatkins**

### CODE REVIEW

**Review by kwryankrattiger — DISMISSED** [review]

File: var/spack/repos/builtin/packages/silo/package.py [hunk 1]
```diff
@@ -68,8 +68,10 @@ class Silo(AutotoolsPackage):
     # autoconf-archive in 2011
     patch('configure-AX_CHECK_COMPILE_FLAG.patch')
 
-    # API changes in 1.13 cause breakage
-    conflicts('hdf5@1.13:', when='+hdf5')
+    # API changes in hdf5-1.13 cause breakage
+    # See https://github.com/LLNL/Silo/pull/260
+    patch('hdf5-113.patch', when='@4.11:')
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/silo/package.py]: Make this conditional on the newer hdf5? Better not to patch code if we can help it

**Review by sethrj — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/silo/package.py (hunk 1 above)
Comment (sethrj) [reply var/spack/repos/builtin/packages/silo/package.py]: (To clarify: if the patch is unconditional, it'll invalidate every current installation of silo.)

```suggestion
    patch('hdf5-113.patch', when='@4.11: ^hdf5@1.13:')
```

**Review by sethrj — DISMISSED** [review]
Thanks!

**Comment by chuckatkins** [comment]
The updated patch just now removed some unrelated white space changes I accidentally let creep in.  I spoke with @brtnfld from The HDF Group and confirmed that this is a valid approach and works to address the API compatibility issues.  (We spoke directly but he also added his approval over on the upstream PR in https://github.com/LLNL/Silo/pull/260)

**Review by kwryankrattiger — APPROVED** [review]

**Pull request merged**